### ISSUE TITLE
How to compile a simple hellow world app?

### ISSUE THREAD

**Issue by aykaramba** [wadehennessey/wcl#9]
I have WCL running, kinda fun.   I am trying to create a simple hello world and compile it to a binary to test.

I have tried:

(defun hello (x) (princ "hello world"))
(compile 'hello)

But I get the following error:

> (compile 'hello)
/tmp/424268-1compile.wcl:14:10: fatal error: lisp.h: No such file or directory
   14 | #include "lisp.h"
      |          ^~~~~~~~
compilation terminated.
Error: Failed to compile file: "/tmp/424268-1compile.wcl", status: 256

0: Abort to level 0 (Name: ABORT) 
WCL: Ignoring signal number 2, aborting to top-level

Obviously I don't know how to compile in WCL and I just tried the compile form based on previous posts.  SBCLs save-lisp-and-die doesn't exist.

How is a WCL program compiled to a binary?

**Comment by wadehennessey** [comment]
Try installing wcl first. -wade

https://github.com/wadehennessey/wcl/blob/master/INSTALL

**Comment by aykaramba** [comment]
Allright.

1. For a fresh / repeatable install I cloned the repo.
2. I updated the Make file for Debian 11 (the lib64 path is different) and installed your precompiled binaries.
3. I tried to compile the function above, I got an error.
4. I thought that the issue might be with the precompiled binary so I compiled WCL successfully.
5. I re-ran make install which should have installed binaries compiled for my system.
6. When I try to compile the above function I get the following error:

```
> (compile 'hello)
Error: #<SIMPLE-VECTOR BIT 72057026716368896 7FF88F1DF105> is not a number

0: Abort to level 0 (Name: ABORT) 
WCL: Ignoring signal number 2, aborting to top-level
```

Basically, I am able to now clone the repo, install the software and recompile without error WCL and reinstall.  Everything should be working as expected and I have a reproducible build process.

Any thoughts on what the solution to the error above might be?

**Comment by aykaramba** [comment]
Closing this issue.  I figured out how to compile with WCL.  This is amazing.

**Comment by vindarel** [comment]
@aykaramba Great to read your progress :rocket:  Are there details you didn't note on the previous comment to fix the `compile` error?

**Comment by aykaramba** [comment]
Yes, there a few things:

1) As per above, you have to update the make file to point to the right paths on your system.  

2) On my Debian machine, I ALSO had to create the following symlink to point to where the installer put the file (I guess maybe it should have been installed in /lib or something, did not double check): libcl.info -> /usr/lib/x86_64-linux-gnu/libcl.info

3) Then you read the docs here: https://github.com/wadehennessey/wcl/blob/master/doc/wcl.text 

Here is an example of how to compile a tiny binary from the docs:

``` 


LINKING APPLICATIONS WITH SHARED LIBRARIES
------------------------------------------

Here's a quick example of how to compile and link an application
consisting of two files:

file bar.lisp:
---------------------
(in-package "USER")

(defstruct point x y)
---------------------

file foo.lisp:
---------------------
(in-package "USER")

(defun lmain ()
  (format t "Point: ~S~%" (make-point :x 3 :y 4)))
---------------------

In WCL:

> (compile-file "bar")
Compiling file #P"bar.lisp"
Wrote object file #P"bar.o"
#P"bar.o"

> (compile-file "foo")
Compiling file #P"foo.lisp"
Wrote object file #P"foo.o"
#P"foo.o"

>  (link-executable '("foo" "bar"))
0 seconds: Reading symbol information
1 seconds: Compiling predicates file
Compiling file #P"/n/kobold/usr1/home/wade/tmp/26123-23preds.lisp"
Wrote object file #P"/n/kobold/usr1/home/wade/tmp/26123-23preds.o"
4 seconds: Writing data file
4 seconds: Compiling data file
10 seconds: done
#P"foo"
>

Now we can run it from a shell:

/home/wade/test> strip foo
/home/wade/test> size foo
text	data	bss	dec	hex
40960	8192	0	49152	c000
/home/wade/test> foo
Point: #<POINT 64ADD>
/home/wade/test> 

```

Those instructions are in the documentation file and it generates a binary that runs.  However, if you pay attention clx has a bug in it and is broken so you won't be able to play too much with the test files.  

If you get a chance, can you test out and document how to interact with the shell, say echo to stdout or something?  I was not able to get that working yet.
